pkg/config/vsphereuser.go

```diff
@@ -0,0 +1,82 @@
+package config
+
+import (
+	_ "embed"
+	"os"
+)
+
+const (
+	EksavSphereUsernameKey = "EKSA_VSPHERE_USERNAME"
+	EksavSpherePasswordKey = "EKSA_VSPHERE_PASSWORD"
+	// Username and password for cloud provider
+	EksavSphereCPUsernameKey = "EKSA_VSPHERE_CP_USERNAME"
+	EksavSphereCPPasswordKey = "EKSA_VSPHERE_CP_PASSWORD"
+	// Username and password for the CSI driver
+	EksavSphereCSIUsernameKey = "EKSA_VSPHERE_CSI_USERNAME"
+	EksavSphereCSIPasswordKey = "EKSA_VSPHERE_CSI_PASSWORD"
+)
+
+type VSphereUserConfig struct {
+	EksaVsphereUsername    string
+	EksaVspherePassword    string
+	EksaVsphereCPUsername  string
+	EksaVsphereCPPassword  string
+	EksaVsphereCSIUsername string
+	EksaVsphereCSIPassword string
+}
+
+//go:embed static/globalPrivs.json
+var VSphereGlobalPrivsFile string
+
+//go:embed static/eksUserPrivs.json
+var VSphereUserPrivsFile string
+
+//go:embed static/adminPrivs.json
+var VSphereAdminPrivsFile string
+
+//go:embed static/cnsDatastorePrivs.json
+var VSphereCnsDatastorePrivsFile string
+
+//go:embed static/cnsSearchAndSpbmPrivs.json
+var VSphereCnsSearchAndSpbmPrivsFile string
+
+//go:embed static/cnsVmPrivs.json
+var VSphereCnsVmPrivsFile string
+
+//go:embed static/cnsHostConfigStorage.json
+var VSphereCnsHostConfigStorageFile string
+
+//go:embed static/readOnlyPrivs.json
+var VSphereReadOnlyPrivs string
+
+func NewVsphereUserConfig() *VSphereUserConfig {
+	eksaVsphereUsername := os.Getenv(EksavSphereUsernameKey)
```

Comment:
Do we want to do validations of whether they are empty or not, or will the govmomi api handle it?

Reply:
already in place 👍 
https://github.com/aws/eks-anywhere/pull/2907#discussion_r949279785